packages/frontend/src/components/Charts/FundingChart.tsx

```diff
@@ -0,0 +1,116 @@
+import { createStyles, makeStyles } from '@material-ui/core'
+import dynamic from 'next/dynamic'
+import React, { useState } from 'react'
+import CustomSwitch from '@components/CustomSwitch'
+import { useNormHistory } from '@hooks/useNormHistory'
+import { NormHistory } from '../../types/index'
+import { graphOptions } from '../../constants/diagram'
+
+const Chart = dynamic(() => import('kaktana-react-lightweight-charts'), { ssr: false })
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    switchWrapper: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      flexWrap: 'wrap',
+    },
+    legendBox: {
+      display: 'flex',
+      gap: '10px',
+      marginTop: '10px',
+      justifyContent: 'center',
+    },
+    legendContainer: {
+      display: 'flex',
+      gap: '5px',
+    },
+  }),
+)
+
+const FundingChart = () => {
+  const fundingTypes = [
+    { id: 'funding', text: 'Funding' },
+    { id: 'vol', text: 'VOL' },
+  ]
+  const fundingDurations = [
+    { id: '1d', text: 'Day' },
+    { id: '1m', text: 'Month' },
+    { id: '1y', text: 'Annual' },
+  ]
+  const [fundingType, setFundingType] = useState(fundingTypes[0])
+  const [fundingDuration, setFundingDuration] = useState(fundingDurations[0])
+  const classes = useStyles()
+
+  const normFactors: NormHistory[] = useNormHistory()
+  const graphData = normFactors.map((item) => {
+    const secondsElapsed = Number(item.timestamp) - Number(item.lastModificationTimestamp)
+    const deltaT = secondsElapsed / (420 * 60 * 60)
+    const markIndex = 1 / Math.exp(Math.log(Number(item.newNormFactor) / Number(item.oldNormFactor)) / deltaT)
+    const dayFunding = Math.log(markIndex) / 17.5
+    const monthFunding = dayFunding * 30
+    const yearFunding = dayFunding * 365.25
```

Comment:
wait lol is it 365.25 to account for leap years?

Reply:
Yes, I think so. Joe calculated yearly funding in that way in the doc.